python/cudf_polars/cudf_polars/dsl/ir.py

```diff
@@ -947,6 +944,7 @@ def do_evaluate(
         # TODO: uniquify
         requests = []
         replacements: list[expr.Expr] = []
+        agg_infos = [req.collect_agg(depth=0) for req in agg_requests]
```

Comment:
This is definitely a good way to avoid `AggInfo` serialization. The disadvantage is that we don't raise an error from `collect_agg` when the `GroupBy` object is created at translation time.